fix(minifier): don't inline single use variable in conditional logical expressions

sapphi-red opened the pull request:
https://github.com/oxc-project/oxc/blob/358f2fccd3183b7987332cb321b946fe31a85691/crates/oxc_minifier/src/peephole/minimize_statements.rs#L1427-L1429
This should be handled in logical expressions as well, but was not handled.

Boshen commented:
### Merge activity

* **Oct 1, 4:55 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

sapphi-red committed:
fix(minifier): don't inline single use variable in conditional logical expressions (#14185)

https://github.com/oxc-project/oxc/blob/358f2fccd3183b7987332cb321b946fe31a85691/crates/oxc_minifier/src/peephole/minimize_statements.rs#L1427-L1429
This should be handled in logical expressions as well, but was not handled.
